lib/services/auth/navigation_service.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+class NavigationService {
+  final GlobalKey<NavigatorState> navigatorKey = GlobalKey<NavigatorState>();
+
+  Future<dynamic> navigateTo(String routeName) {
+    return navigatorKey.currentState.pushNamed(routeName);
+  }
+
+  void displayError(dynamic error) {
+    final message = error.toString();
+
+    // filter out messages user doesnt want to see here
+
+    // show dialog
+    final newContext = navigatorKey.currentState.overlay.context;
+    showDialog<dynamic>(
```

Comment:
One thing that I forgot about when suggesting this strategy is that calling showDialog either during a build or not during a build... I can't remember the details but there can be weird subtle issues.  I've got an issue tracking working on it: https://github.com/nickmeinhold/crowdleague/issues/110  

I think it just needs a little beefing up to deal with all scenarios, again I reckon don't worry about it now and I'll let you know what I come up with when I get that issue done.